spring-kafka/src/main/java/org/springframework/kafka/listener/KafkaMessageListenerContainer.java

```diff
@@ -372,8 +379,41 @@ public void onPartitionsAssigned(Collection<TopicPartition> partitions) {
 				consumer.assign(new ArrayList<>(this.definedPartitions.keySet()));
 			}
 			this.consumer = consumer;
-			this.listener = listener;
-			this.acknowledgingMessageListener = ackListener;
+			Object theListener = listener == null ? ackListener : listener;
```

Comment:
That might not make so much sense, but I'm changing this variable to the `KafkaDataListener<?>` for better code tracability.